src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateExecution.java

```diff
@@ -94,28 +90,31 @@ protected ExternalWorkspace run() throws Exception {
                 throw new AbortException(format("'%s' doesn't have any stable build", upstreamName));
             }
 
-            ExwsAllocateActionWrapper exwsActionWrapper = lastStableBuild.getAction(ExwsAllocateActionWrapper.class);
-            if (exwsActionWrapper == null) {
-                String message = format("The Jenkins job '%s' does not have registered any 'External Workspace Allocate' action. Did you run exwsAllocate step in the upstream job?", upstreamName);
+            ExwsAllocateActionImpl allocateAction = lastStableBuild.getAction(ExwsAllocateActionImpl.class);
+            if (allocateAction == null) {
+                String message = format("This downstream job has to allocate the same external workspace used by the upstream job. " +
+                        "To do so, the exwsAllocate step must be called in the upstream job. " +
+                        "Please call the exwsAllocate step in the upstream job, because the build '%s' doesn't have such calls.", lastStableBuild);
```

Comment:
It's better, but I have one mention: if the upstream job doesn't have any stable build, the user will get the error message from line [90](https://github.com/jenkinsci/external-workspace-manager-plugin/pull/8/files#diff-fe1322463bd4bb50e34cee4801cbfd3eR90). Therefore, the first sentence: "No stable build upstream workspace could be found" may be a little bit redundant.

So, IMO it would be enough to say: "The upstream job must have at least one stable build with a call to the exwsAllocate step in order to have a workspace usable by this job."

WDYT?